odoo/cli/populate.py

```diff
@@ -1,131 +1,70 @@
-                        db=env.cr.dbname
-                    )
-                else:
-                    profiling_context = nullcontext()
-
-                if commit:
-                    commit_context = nullcontext()
-                else:
-                    commit_context = patch('odoo.sql_db.Cursor.commit')
-
-                _logger.info('Populating database for model %s', model._name)
-                t0 = time.time()
-
-                with profiling_context, commit_context:
-                    registry.populated_models[model._name] = model._populate(size).ids
-
-                    if not registry.populated_models[model._name]:
-                        # Do not create ir.profile records
-                        # for models without any population factories
-                        profiling_context.db = False
-
-                    # force the flush to make sure population time still
-                    # considers flushing all values to database
-                    env.flush_all()
-
-                if commit:
-                    env.cr.commit()
-
-                model_time = time.time() - t0
-                if model_time > 1:
-                    _logger.info('Populated database for model %s (total: %fs) (average: %fms per record)',
-                                 model._name, model_time, model_time / len(registry.populated_models[model._name]) * 1000)
-        except:
-            _logger.exception('Something went wrong populating database')
-        finally:
-            if not commit:
-                env.cr.rollback()
-            populated_models = registry.populated_models
-            del registry.populated_models
-
-        return populated_models
-
-    @classmethod
-    def _get_ordered_models(cls, env, model_patterns=False):
-        _logger.info('Computing model order')
-        processed = set()
-        ordered_models = []
-        visited = set()
-        def add_model(model):
-            if model not in processed:
-                if model in visited:
-                    raise ValueError('Cyclic dependency detected for %s' % model)
-                visited.add(model)
-                for dep in model._populate_dependencies:
-                    add_model(env[dep])
-                ordered_models.append(model)
-                processed.add(model)
-        for model in env.values():
-            if model_patterns and not any(fnmatch.fnmatch(model._name, match) for match in model_patterns):
-                continue
-            if model._transient or model._abstract:
-                continue
-            ir_model = env['ir.model'].search([('model', '=', model._name)])
-            if not model_patterns and all(module.startswith('test_') for module in ir_model.modules.split(',')):
-                continue
-            add_model(model)
-
-        return ordered_models
+    def populate(cls, env: Environment, factors: Iterator[int], model_names: Iterator[str], separator_code: int):
```

Comment:
`factors` is a list.  A list is not an _iterator_, but it is an _iterable_ :wink: 
```suggestion
    def populate(cls, env: Environment, factors: Iterable[int], model_names: Iterable[str], separator_code: int):
```

Reply:
Yeah ... that's me being stupid, tnx for the catch :)